iceoryx_hoofs/test/moduletests/test_cxx_string.cpp

```diff
@@ -2214,6 +2305,113 @@ TEST(String10, UnsafeAppendTooLargeStringLiteralFails)
     EXPECT_THAT(testString.c_str(), StrEq("Kern"));
 }
 
+TYPED_TEST(stringTyped_test, UnsafeAppendNullCharWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "b1a6743f-1146-4269-9c13-0baa1e14fbe4");
+    using MyString = typename TestFixture::stringType;
+    constexpr auto STRINGCAP = MyString::capacity();
+    string<STRINGCAP + 1U> testString("M");
+    const char testChar = '\0';
+    EXPECT_THAT(testString.unsafe_append(testChar), Eq(true));
+    EXPECT_THAT(testString.capacity(), Eq(STRINGCAP + 1U));
```

Comment:
We should drop this capacity check. There are other places where this also applies IMO. Whenever we create the string of a type with the exact capacity we check for we basically reinvent the `capacity` function test.